taker/src/routes.rs

```diff
@@ -159,12 +167,17 @@ pub struct CfdOrderRequest {
 }
 
 #[rocket::post("/cfd/order", data = "<cfd_order_request>")]
-#[instrument(name = "POST /cfd/order", skip(taker, _auth), err)]
+#[instrument(name = "POST /cfd/order", skip(taker, user), err)]
 pub async fn post_order_request(
     cfd_order_request: Json<CfdOrderRequest>,
     taker: &State<Taker>,
-    _auth: Authenticated,
+    user: User,
 ) -> Result<(), HttpApiProblem> {
+    if user.expired {
+        return Err(HttpApiProblem::new(StatusCode::FORBIDDEN)
+            .title("Password expired")
+            .detail("Please change your password to continue"));
```

Comment:
🤔 I'm not sure I follow the logic. Why should an expired password stop you from using the app? A wrong password, sure. But, as a user, I've come to expect to be able to use accounts even if they're nagging me to change my password for whatever reason.

Also, having to add all these expiry checks on every endpoint is not great. If for some reason we want to stop the user from using the app because their password has expired, shouldn't we just _not_ mount any of these routes?

Reply:
> 🤔 I'm not sure I follow the logic. Why should an expired password stop you from using the app? A wrong password, sure. But, as a user, I've come to expect to be able to use accounts even if they're nagging me to change my password for whatever reason.

As a user yes, but as the app I want to protect you until you have a secure password.

> Also, having to add all these expiry checks on every endpoint is not great. If for some reason we want to stop the user from using the app because their password has expired, shouldn't we just not mount any of these routes?

I'm not sure how we could achieve this in a more general way.